.github/workflows/tests.yml

```diff
@@ -268,46 +250,36 @@ jobs:
         run: |
           tar xf ocaml-build-artifacts.tgz
           sudo cp ocaml-build-artifacts/bin/* /usr/bin
-      - name: Install semgrep
+      - name: Install Python dependencies
+        working-directory: semgrep
+        run: pipenv install --dev
+      - name: Run pytest
+        working-directory: semgrep
         run: |
-          cd semgrep
-          pip install pipenv==2021.5.29
-          pipenv install --dev
-      - name: Test semgrep
-        run: |
-          cd semgrep
-
           # tests should simulate CI environment iff they need one
           unset CI
           unset "${!GITHUB_@}"
 
-          make check-markers
-          pipenv run pytest --ignore=tests/qa/test_public_repos.py -n auto -vv --tb=short --durations=0 --snapshot-update tests/
-      - name: Generate cheatsheets for semgrep.dev
-        run: ./scripts/generate-cheatsheet
-      - name: Upload artifacts
-        uses: actions/upload-artifact@v3
-        with:
-          name: cheatsheet
-          path: cheatsheet.json
+          pipenv run pytest -n auto -vv --snapshot-update --allow-snapshot-deletion
+      # because of the fail-fast setting, we expect only the fastest failing job to get to the steps below
       - name: Prepare repo for snapshot commit
-        if: ${{ failure() }}
+        if: failure()
         run: |
           # the commit step that follows will fail to fetch the pfff submodule
           # (perhaps because of the github token's permissions)
           # so we disable recursive fetching
           git config fetch.recurseSubmodules false
       - name: Commit snapshot updates
         id: snapshot-commit
-        if: ${{ failure() && github.event_name == 'pull_request' }}
+        if: failure() && github.event_name == 'pull_request'
```

Comment:
Might be worth adding a guard not to do this if they dont have permissions (from a fork) so the failing tests dont get eclipsed by the error message saying they dont have perms

Reply:
Can do in follow up

Reply:
```suggestion
        if: failure() && github.event_name == 'pull_request' && (github.actor != 'dependabot[bot]' && !(github.event.pull_request.head.repo.full_name != github.repository))
```